src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+    gDPSetCombineLERP(POLY_XLU_DISP++, TEXEL0, PRIMITIVE, PRIM_LOD_FRAC, TEXEL0, TEXEL0, 0, PRIMITIVE, 0, PRIMITIVE,
+                      ENVIRONMENT, COMBINED, ENVIRONMENT, 0, 0, 0, COMBINED);
+    gDPSetRenderMode(POLY_XLU_DISP++, G_RM_PASS, G_RM_ZB_CLD_SURF2);
+    gSPClearGeometryMode(POLY_XLU_DISP++, G_FOG | G_LIGHTING | G_TEXTURE_GEN | G_TEXTURE_GEN_LINEAR);
+    gSPSetGeometryMode(POLY_XLU_DISP++, G_ZBUFFER | G_SHADE | G_SHADING_SMOOTH);
+    gDPPipeSync(POLY_XLU_DISP++);
+
+    gDPSetEnvColor(POLY_XLU_DISP++, this->altEnvColor.r, this->altEnvColor.g, this->altEnvColor.b, this->altEnvColor.a);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void (*sSetupHandlers[])(GraphicsContext* gfxCtx, EffectBlure* this, Vtx* vtx) = {
+    EffectBlure_SetupSimple,
+    EffectBlure_SetupSimpleAlt,
+};
+
+s32 D_801AE268 = 0; // unused
+
+void EffectBlure_DrawSimpleVertices(GraphicsContext* gfxCtx, EffectBlure* this, Vtx* vtx) {
+    Mtx* mtx;
+
+    OPEN_DISPS(gfxCtx);
+
+    sSetupHandlers[this->drawMode](gfxCtx, this, vtx);
+    gDPPipeSync(POLY_XLU_DISP++);
+
+    {
+        Vec3f sp1B0;
+        Vec3f sp1A4;
+        Vec3f sp198;
+        f32 alphaRatio;
+        MtxF sp154;
+        MtxF sp114;
+        MtxF spD4;
+        MtxF sp94;
+        f32 scale;
+        s32 i;
+        s32 j;
+
+        j = 0;
+
+        for (i = 0; i < this->numElements - 1; i++) {
+            if (this->drawMode == 1) {
+                alphaRatio = (f32)this->elements[i].timer / (f32)this->elemDuration;
+                gDPSetPrimColor(POLY_XLU_DISP++, 0x00, 0x80, this->altPrimColor.r, this->altPrimColor.g,
+                                this->altPrimColor.b, this->altPrimColor.a * (1.0f - alphaRatio));
+                gDPPipeSync(POLY_XLU_DISP++);
+            }
+
+            if (1) {} // Necessary to match
+
+            gSPVertex(POLY_XLU_DISP++, &vtx[j], 4, 0);
+            gSP2Triangles(POLY_XLU_DISP++, 0, 1, 3, 0, 0, 3, 2, 0);
+
+            if (this->flags & 4) {
+                sp1B0.x = ((f32)vtx[4 * i + 0].v.ob[0] + (f32)vtx[4 * i + 1].v.ob[0]) * 0.5f;
+                sp1B0.y = ((f32)vtx[4 * i + 0].v.ob[1] + (f32)vtx[4 * i + 1].v.ob[1]) * 0.5f;
+                sp1B0.z = ((f32)vtx[4 * i + 0].v.ob[2] + (f32)vtx[4 * i + 1].v.ob[2]) * 0.5f;
+                sp1A4.x = ((f32)vtx[4 * i + 2].v.ob[0] + (f32)vtx[4 * i + 3].v.ob[0]) * 0.5f;
+                sp1A4.y = ((f32)vtx[4 * i + 2].v.ob[1] + (f32)vtx[4 * i + 3].v.ob[1]) * 0.5f;
+                sp1A4.z = ((f32)vtx[4 * i + 2].v.ob[2] + (f32)vtx[4 * i + 3].v.ob[2]) * 0.5f;
+
+                Math_Vec3f_Diff(&sp1A4, &sp1B0, &sp198);
+                scale = sqrtf(SQ(sp198.x) + SQ(sp198.y) + SQ(sp198.z));
+
+                if (fabsf(scale) > 0.0005f) {
+                    scale = 1.0f / scale;
+                    Math_Vec3f_Scale(&sp198, scale);
+
+                    SkinMatrix_SetTranslate(&sp154, sp1B0.x, sp1B0.y, sp1B0.z);
+                    SkinMatrix_SetRotateAroundVec(&sp114, 0x3FFF, sp198.x, sp198.y, sp198.z);
```

Comment:
No suggestion here, but an observation that for 90 degree and 180 degree angles, OoT commonly has those angles defined as `0x3FFF` and `0x7FFF` respectively, while MM has been very consistent in replacing those with angles `0x4000` and `0x8000` respectively. This seems like a rare case in MM where `0x3FFF` still remains. Figured that was worth commenting on